presto-main/src/main/java/com/facebook/presto/execution/SqlQueryExecution.java

```diff
@@ -356,7 +360,10 @@ public int getRunningTaskCount()
     @Override
     public void start()
     {
-        try (SetThreadName ignored = new SetThreadName("Query-%s", stateMachine.getQueryId())) {
+        try (SetThreadName ignored = new SetThreadName("Query-%s", stateMachine.getQueryId());
+                ThreadWatchdog w = new ThreadWatchdog(Thread.currentThread(),
```

Comment:
I think it's less risky to just scope this to the query analyzer and not all the other methods including starting the scheduler.

That said there might be value in being able to interrupt the entire thing. @highker WDYT?

Reply:
I assumed the thread may get stuck anywhere, not just the specific discovered case. But considering I named the property `query_analyzer_timeout`, it probably makes sense to wrap only the analyzer part.